grin/src/Pipeline/Pipeline.hs

```diff
@@ -346,6 +346,7 @@ saveTransformationInfo name content = do
   outputDir <- view poOutputDir
   let fname = printf "%03d.%s" n name
   liftIO $ do
+    createDirectoryIfMissing True outputDir
```

Comment:
This is now moved into the runPipeline: https://github.com/grin-tech/grin/blob/master/grin/src/Pipeline/Pipeline.hs#L574

Reply:
Since `SaveGrin` can be called from anywhere,  I think we have to make sure it cannot fail due to missing directory exceptions. `runPipeline` is a really big global function, and the user might not want to call it simply to save the result of a transformation. 

As an example, the optimization pass (`--optimize`) needs to save the resulting code at each step. It uses the `SaveGrin` pipelinestep for this.

Reply:
The runPipeline should be the only entry point to evaluate any Pipeline computation, thus SaveGrin shouldn't be called anywhere else just within a created pipeline computation.

The runPipeline funcion isn't that big and its responsibility to set up the computational context appropriately.